Listener Management Improvements

Vampire opened the pull request:
- Remove unnecessary class parameters for listener registration methods
- Do not register listeners multiple times to the same context and always use the same listener manager, which now also gets its remove handlers invoked whenever the listener is removed from its context, not only when removed through the listener manager
- Add DiscordApi#removeObjectListener and DiscordApi#removeListener

Vampire commented:
Ok, totally changed what I started to do, but now it is ready for review. I hope I did all correct, especially for the docs, as there were some renamings like `...AttachableListener` and so on were involved. Feel free to complain if something is inconsistent. :-)

BtoBastian commented on src/main/java/de/btobastian/javacord/entities/message/Message.java:
```diff
@@ -353,6 +360,612 @@
+        try {
+            return addMessageDeleteListener(api, Long.valueOf(messageId), listener);
+        } catch (NumberFormatException ignored) {
+            return null;
```
I don't think returning null is the right approach here.
I remember running into the same issue when creating the `ListenerManager` class.
Iirc I just removed the methods with a `String` parameter in the places where it would cause these kind of issues.
Another approach would be to just throw an `IllegalArgumentException`.

Vampire replied:
However you like it better.
I just tried to follow how it was done for others.
If you want an IAE, we can just remove the try/catch, as NFE is an IAE.
The question is, whether this is too inconsistent then, as some methods return a default value (empty list) and some throw an exception when the string is not a proper number.

BtoBastian replied:
I think it would be best to just removed these methods.

Vampire replied:
ok, will do

Vampire commented:
New push available:
- change `newName` and `newColor` to be fields of the respective role events
- fixed the formatting of the JavaDoc of `ServerMemberJoinListner`
- added the multi-listener magic for the other channels that don't have attachable events right now for easy addition in the future
- removed the null-returning `Message` methods
